subscription-engine/kafka-ws/app/Redis/RedisManager.js

```diff
@@ -21,6 +22,22 @@ class RedisManager {
     // TODO: Implement "retry_strategy"
     this.redisClient = redis.createClient(this.config);
     logger.info('RedisManager singleton creation complete!');
+
+    const stateService = 'redis';
+    StateManager.registerService(stateService);
+    this.redisClient.on('connect', () => StateManager.signalReady(stateService));
+    /**
+     * The 'error' event must be mapped, otherwise the application hangs on an uncaughtException
+     * and some unexpected behaviors happens.
+     *
+     * The error event doesn't mean the service is unhealthy, because it can be
+     * AbortError, ParserError AggregateError, or others subclasses of RedisError
+     * When the client disconnects to redis the 'end' event is fired, there we can consider
+     * the service is unhealthy
+     */
+    this.redisClient.on('error', (error) => logger.warn(`${error}`));
```

Comment:
When is this error event triggered?

Reply:
This event is triggered when encountering an error connecting to the Redis server or when any other in Node Redis occurs.